transformer/transformer.go

```diff
@@ -26,43 +26,43 @@ type contentItem struct {
 	Description description `json:"description"`
 	Type        string      `json:"type"`
 	URI         string      `json:"uri"`
-	Matches     matches     `json:"matches,omitempty"`
+	Matches     *matches    `json:"matches,omitempty"`
 }
 
 type description struct {
-	Summary           string   `json:"summary"`
-	NextRelease       string   `json:"next_release,omitempty"`
-	Unit              string   `json:"unit,omitempty"`
-	Keywords          []string `json:"keywords,omitempty"`
-	ReleaseDate       string   `json:"release_date,omitempty"`
-	Edition           string   `json:"edition,omitempty"`
-	LatestRelease     bool     `json:"latest_release,omitempty"`
-	Language          string   `json:"language,omitempty"`
-	Contact           contact  `json:"contact,omitempty"`
-	DatasetID         string   `json:"dataset_id,omitempty"`
-	Source            string   `json:"source,omitempty"`
-	Title             string   `json:"title"`
-	MetaDescription   string   `json:"meta_description,omitempty"`
-	NationalStatistic bool     `json:"national_statistic,omitempty"`
-	Headline1         string   `json:"headline1,omitempty"`
-	Headline2         string   `json:"headline2,omitempty"`
-	Headline3         string   `json:"headline3,omitempty"`
+	Summary           string    `json:"summary"`
+	NextRelease       *string   `json:"next_release,omitempty"`
+	Unit              *string   `json:"unit,omitempty"`
```

Comment:
There is no need to set a pointer to string type, the omitempty flag will remove an empty string `""` from the json response.

See demonstration below:

```
package main

import (
	"fmt"
	"encoding/json"
	"os"
)

type Dummy struct {
        String    string  `json:"name,omitempty"`
        Number  int64   `json:"number,omitempty"`
        Pointer *string `json:"pointer,omitempty"`
}

func main() {
       // These are considered to be empty by go
       dummyNilPointer := Dummy{
                String:    "",
                Number:  0,
                Pointer: nil,
        }

        dataNil, err := json.Marshal(dummyNilPointer)
        if err != nil {
                fmt.Printf("An error occured: %v", err)
                os.Exit(1)
        }

        fmt.Println(string(dataNil))
}
```

Output:
`{}`

A good blogpost regarding how the field type and json tags work for `json.Marshal` or `json.Unmarshal`:  https://vsupalov.com/go-json-omitempty/

Reply:
Basically all fields which are of type string should not be set to a pointer, same would go for integer values.

The only scenario where you would make them pointers is when an empty string is a valid value of a field and 0 is a valid integer value of a filed (e.g. count, number_of_items etc.)

Reply:
As @nshumoogum has said, strings and integers do not need to be pointers, just the structs, arrays and booleans.